darts/models/forecasting/foundation_model.py

```diff
@@ -0,0 +1,402 @@
+            if not path.exists():
+                raise FileNotFoundError(
+                    f"File {filename} not found in {self.local_dir}"
+                )
+            if not path.is_file():
+                raise ValueError(f"Path {path} is not a file")
+            return path
+        else:
+            file_path = hf_hub_download(
+                repo_id=self.repo_id,
+                filename=filename,
+                revision=self.repo_commit,
+            )
+            return Path(file_path)
+
+    def _load_config(
+        self,
+    ) -> dict:
+        """Load the model configuration from a JSON file.
+
+        Returns
+        -------
+        dict
+            The model configuration.
+        """
+        config_path = self._get_file_path(self.config_file)
+        with open(config_path) as f:
+            config = json.load(f)
+        return config
+
+    def _load_model_weights(
+        self,
+        module: PLForecastingModule,
+    ) -> None:
+        """Load the model weights from a safetensors file.
+
+        Parameters
+        ----------
+        module
+            The PyTorch module to load the weights into.
+        """
+        module_path = self._get_file_path(self.model_file)
+        state_dict = load_file(module_path)
+        module.load_state_dict(state_dict)
+
+    @staticmethod
+    def _extract_module_params(
+        module_class: type[PLForecastingModule],
+        config: dict,
+    ):
+        """Extract params from `config` to set up the given `module_class`."""
+        get_params = list(inspect.signature(module_class.__init__).parameters.keys())
+        get_params.remove("self")
+        return {kwarg: config.get(kwarg) for kwarg in get_params if kwarg in config}
+
+    def _load_model(
+        self,
+        module_class: type[PLForecastingModule],
+        pl_module_params: dict,
+        additional_params: dict = {},
+    ) -> PLForecastingModule:
+        """Load the model by creating an instance of the given module class and loading
+        the weights. Some configuration files might contain external parameters that
+        are not part of the module class constructor like `architectures`. They are filtered
+        out before instantiating the module.
+
+        Parameters
+        ----------
+        module_class
+            The class of the PyTorch Lightning module to instantiate.
+
+        Returns
+        -------
+        PLForecastingModule
+            The loaded PyTorch Lightning module.
+        """
+        config = self._load_config()
+        module_params = self._extract_module_params(module_class, config)
+        module = module_class(
+            **module_params,
+            **pl_module_params,
+            **additional_params,
+        )
+        self._load_model_weights(module)
+        return module
+
+
+class FoundationModel(MixedCovariatesTorchModel):
+    _allows_finetuning: bool = False
+    _probabilistic: bool = False
```

Comment:
We can probably infer everything under the hood dynamically, using `likelihood` and `TorchForecastingModel.supports_probabilistic_prediction`

```suggestion
```